### PR TITLE
doc: We support Postgres-CDC with REPLICA IDENTITY DEFAULT  

### PR DESCRIPTION
**Pull request by msudberg**
### Motivation

  * This PR fixes a recognized bug. [#6465 ]

### Description

Our CDC-Postgres guide specified that you had to set Postgres to `REPLICA IDENTITY FULL`, which can cause a performance hit. We now support `REPLICA IDENTITY DEFAULT` as well.

### Checklist

- [x] This PR has adequate test coverage.
- [ ] This PR adds a release note for any user-facing behavior changes.

### CODE REVIEW

File: doc/user/content/third-party/debezium.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Debezium emits records using an envelope that contains valuable information abou
 Currently, Materialize only supports Avro-encoded Debezium records. If you're interested in JSON support, please reach out in the community Slack or leave a comment on [this GitHub issue](https://github.com/MaterializeInc/materialize/issues/5231).
 {{</ note >}}
 
-### Database settings
+## Database settings
```
Comment (morsapaes) [inline doc/user/content/third-party/debezium.md:18]: Maybe it makes sense to just direct people to the CDC guides here. I'm going to push the MySQL one in the next few days, so I can also revamp this page as part of that PR.

File: doc/user/content/guides/cdc-postgres.md [hunk 2]
```diff
@@ -46,12 +48,12 @@ Before creating a source in Materialize, you need to ensure that the upstream da
 1. Set the replica identity value for the tables you want to replicate:
 
     ```sql
-    ALTER TABLE repl_table REPLICA IDENTITY FULL;
+    ALTER TABLE repl_table REPLICA IDENTITY DEFAULT;
```
Comment (morsapaes) [inline doc/user/content/guides/cdc-postgres.md:51]: I'm not entirely sure that this is correct for the direct Postgres source. To use `DEFAULT` the source has to work in upsert mode and the source tables would have to have primary keys defined, so unless this was tested I'd like to double check the behaviour.

File: doc/user/content/third-party/debezium.md [hunk 3]
```diff
@@ -1,36 +0,0 @@
----
```
Comment (morsapaes) [inline doc/user/content/third-party/debezium.md:1]: I think we could have kept this page, just rearranged the content around a bit.

Comment (msudberg) [reply doc/user/content/third-party/debezium.md:1]: We could restore it, but it doesn't have any information that isn't in the guide, I don't think.

**Pull request merged**